fastlane/Fastfile

```diff
@@ -238,12 +252,39 @@ end
       project_slug: 'simplenote-macos',
       dsym_path: dSYM_PATH,
     )
-   
+  end
+
+  lane :build_and_upload_app_store do | options |
+    sync_code_signing(
+      type: 'appstore',
+      platform: 'macos',
+      # Will default to false, unless a readonly value is specified, or is
+      # running on CI, in which case it will always be true.
+      readonly: options[:readonly] || is_ci,
+      app_identifier: 'com.automattic.SimplenoteMac'
+    )
+
+    archive_path = File.join(@BUILD_FOLDER, 'Simplenote-Mac.xcarchive')
+    build_simplenote(codesign: true, archive_path: archive_path)
+
+    archive_zip_path = "#{archive_path}.zip"
+    zip(path: archive_path, output_path: archive_zip_path)
+
+    # TODO: Upload to App Store Connect
+
+    # TODO: Add "create_release" parameter to opt out from here
```

Comment:
These TODOs will obviously be addressed before this gets out of the Draft state.